new scraper

emilyo9264 opened the pull request:
Hey Xiyang, I've been working on the scraper and this is what I have so far.

The firstTime function pulls 9999 records from arXiv to create a JSON file (I'm just experimenting with json for now, haven't tried out parquet or sql yet) (FYI I've also included the json file in the folder).

The update function checks if there's any new article published. I'm not sure how to delete individual lines within a file (when there's a new version), so my current solution is to rewrite the file for now. Could you please take a look at it and see if there's any mistake/how I can improve on it? Thx!

settinghead commented:
Hi @emilyo9264, I tried to run your code but got the following error:
```
Updated the dictionary.
Added one new article.
Updated the dictionary.
Added one new article.
Traceback (most recent call last):
  File "DeepLearningPaperScraper.py", line 75, in <module>
    updateArticles()
  File "DeepLearningPaperScraper.py", line 61, in updateArticles
    article_index = orderedArticles.loc[orderedArticles['id'] == item].index[0]
  File "/Users/settinghead/anaconda3/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2084, in __getitem__
    return getitem(key)
IndexError: index 0 is out of bounds for axis 0 with size 0
```

Can you see if you are able to reproduce it?

I suspect it might have to do with the way ID is parsed. Maybe some author updated their article to a new version during the past 14 hours? I'm not sure. 

BTW for now to simplify things I think it's OK to treat different versions of the paper as separate entities (i.e. `xxxxxxv1` would be a different paper from `xxxxxxxv2`.

settinghead commented on paper-collector/DeepLearningPaperScraper.py:
```diff
@@ -0,0 +1,75 @@
+                 'arxiv_primary_category', 'published', 'summary', 
+                 'tags', 'updated'])
+
+    temp_id_2 = ordered_new_artiDF['id'].tolist()
```
Could you rename `temp_id_2` to something more meaningful?

emilyo9264 commented:
Hi Xiyang, here are some changes I made:

1. The id issue was indeed an error - I've fixed that and now it's able to replace the old version with the new version (I figured we should avoid having outdated versions take up extra storage space as #records grows larger and larger);
2. I've added the article_id function to clean the logic up;
3. I've reviewed the style guide and changed some variable names;
4. The "json dumps" seems to be causing issues when I try to reload the file. I'll need to look into that more.

Feel free to let me know if you have more comments/suggestions! Thanks!

settinghead commented:
@emilyo9264 looks great! I added a new commit with a JSON indenter. You may merge any time now. Thanks.
